staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -51,22 +56,26 @@ var (
 // arguments based on user's kuberc configuration.
 type PreferencesHandler interface {
 	AddFlags(flags *pflag.FlagSet)
-	Apply(rootCmd *cobra.Command, args []string, errOut io.Writer) ([]string, error)
+	// ApplyPluginPolicy(*genericclioptions.ConfigFlags)
+	Apply(rootCmd *cobra.Command, kubeConfigFlags *genericclioptions.ConfigFlags, args []string, errOut io.Writer) ([]string, error)
 }
 
 // Preferences stores the kuberc file coming either from environment variable
 // or file from set in flag or the default kuberc path.
 type Preferences struct {
-	getPreferencesFunc func(kuberc string, errOut io.Writer) (*config.Preference, error)
-
-	aliases map[string]struct{}
+	getPreferencesFunc func(kuberc string, errOut io.Writer) (*config.Preference, error) // DefaultGetPreferences
+	aliases            map[string]struct{}
+	pluginPolicy       clientcmdapi.PluginPolicy
```

Comment:
Why do we have `pluginPolicy` in here?. As far as I understand, `applyPluginPolicy` can work without accessing fields in `p Preference`.